bavi/modules/url.py

```diff
@@ -0,0 +1,108 @@
+import logging
+import requests
+from bs4 import BeautifulSoup
+from urllib.parse import urlparse
+import re
+
+log = logging.getLogger('bavi.modules.url')
+
+URL_REGEX = r'(https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=\u263a-\U0001f645]{1,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*))'
+
+IP_REGEX = r"(https?:\/\/([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])(?<!172\.(16|17|18|19|20|21|22|23|24|25|26|27|28|29|30|31))(?<!127)(?<!10)(?<!0)\.([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])(?<!192\.168)(?<!172\.(16|17|18|19|20|21|22|23|24|25|26|27|28|29|30|31))\.([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])(?<!\.255$)([-a-zA-Z0-9@:%_\+.~#?&//=]*))"
+
+IGNORE_REGEX = r'(?i)\.(jpg|jpg:large|png|gif|bmp|tiff|psd|mp4|mkv|avi|mov|mpg|vob|mp3|aac|wav|flac|ogg|mka|wma|zip|rar|7z|tar|iso|exe|dll|cfg|rc|reg|pxe|efi)$'
+
+compiled_url_regex = re.compile(URL_REGEX, re.UNICODE)
+compiled_ip_regex = re.compile(IP_REGEX)
+compiled_ignore_regex = re.compile(IGNORE_REGEX)
+
+TITLE_FORMAT = '[ {0} ] - {1}'
+
+
+def init(bot):
+    bot.add_command('title', title_command)
+    bot.add_matcher(compiled_url_regex, title_command, priority='low')
+    bot.add_matcher(compiled_ip_regex, title_command, priority='low')
+
+
+def title_command(bot, source, target, message, **kwargs):
+    log.debug('url module triggered on: ' + message)
+
+    if len(message) > 0:
```

Comment:
I think a lot of this method can be flattened because your methods implicitly return nothing if the input was blank.

For example, below you have

```
if len(thing) > 0:
  for x in thing:
```

Which is redundant.